rfd/0144-client-tools-updates.md

```diff
@@ -0,0 +1,309 @@
+---
+authors: Russell Jones (rjones@goteleport.com) and Bernard Kim (bernard@goteleport.com)
+state: draft
+---
+
+# RFD 0144 - Client Tools Updates
+
+## Required Approvers
+
+* Engineering: @sclevine && @bernardjkim && @r0mant
+* Product: @klizhentas || @xinding33
+* Security: @reedloden
+
+## What/Why
+
+This RFD describes how client tools (like `tsh` and `tctl`) can be kept up to
+date, either using automatic updates or self-managed updates.
+
+Keeping client tools updated helps with security (fixes for known security
+vulnerabilities are pushed to endpoints), bugs (fixes for resolved issues are
+pushed to endpoints), and compatibility (users no longer have to learn and
+understand [Teleport component
+compatibility](https://goteleport.com/docs/upgrading/overview/#component-compatibility)
+rules).
+
+## Details
+
+### Summary
+
+Client tools (like `tsh` and `tctl`) will automatically download and install
+the version of client tools recommended by the Teleport cluster.
+
+The `cluster_maintenance_config` resource will be updated to include a
+`tools_version` field to control the version of client tools advertised by the
+cluster. This resource will not be configurable on Teleport Cloud. It will be
+managed by the Teleport Cloud team.
+
+If cluster administrators want to opt-out of automatic updates and manage
+updates themselves, they will be able to watch the `cluster_maintenance_config`
+resource for changes and push out updates to endpoints manually.
+
+Inspiration drawn from https://go.dev/doc/toolchain.
+
+### Implementation
+
+#### Client tools
+
+##### Automatic updates
```

Comment:
I think there's the local `tctl` edge case. We should avoid trying to download anything if you're running `tctl` against a local teleport, as you're likely troubleshooting something and downloading potentially wrong version won't help. It's better to use the version from the same packages that installs `teleport`/`teleport-ent`.

Reply:
Sure, that's reasonable, I'll update the RFD.